packages/compile-solidity/src/compilerSupplier/Cache.ts

```diff
@@ -24,11 +26,30 @@ export class Cache {
   add(code: string, fileName: string) {
     const filePath = this.resolve(fileName);
     fs.writeFileSync(filePath, code);
+    this.memoizedCompilers[filePath] = code;
   }
 
   has(fileName: string) {
-    const file = this.resolve(fileName);
-    return fs.existsSync(file);
+    const filePath = this.resolve(fileName);
+    return fs.existsSync(filePath);
+  }
+
+  loadFile(fileName: string): string {
+    const filePath = this.resolve(fileName);
+    if (this.memoizedCompilers.has(filePath)) {
```

Comment:
Can you add a test to verify `memoizedCompilers`, please?

Reply:
done